src/back/glsl.rs

```diff
@@ -251,9 +251,43 @@ pub fn write<'a>(module: &'a Module, out: &mut impl Write, options: Options) ->
             })?
         };
 
-        if let Some(ref binding) = global.binding {
-            if !es {
-                write!(out, "layout({}) ", Binding(binding))?;
+        let format = match module.types[global.ty].inner {
+            TypeInner::Image {
+                class: ImageClass::Storage(format, _),
+                ..
+            } => Some(format),
+            _ => None,
+        };
+
+        if format.is_some() || global.binding.is_some() {
```

Comment:
if it's a storage texture type, it has to have a binding. I think we can omit the `format.is_some()` check